src/state/products.js

```diff
@@ -58,15 +66,24 @@ export function isValidExtractedProduct(extractedProduct) {
     return false;
   }
 
-  for (const key of ['url', 'image']) {
-    try {
-      const url = new URL(extractedProduct[key]);
-      if (!['https:', 'http:'].includes(url.protocol)) {
-        return false;
-      }
-    } catch (err) {
+  try {
+    const url = new URL(extractedProduct.url);
+    const imageUrl = new URL(extractedProduct.image);
+
+    if (!['https:', 'http:'].includes(url.protocol)) {
```

Comment:
Can we move the rest of this logic outside the try block, to avoid catching errors unrelated to URL parsing?